src/ray/raylet/node_manager.cc

```diff
@@ -433,18 +470,37 @@ void NodeManager::ProcessClientMessage(
             });
         RAY_CHECK(running_tasks.size() != 0);
         RAY_CHECK(it != running_tasks.end());
-        JobID job_id = it->GetTaskSpecification().DriverId();
+        const TaskSpecification &spec = it->GetTaskSpecification();
+        const JobID job_id = spec.DriverId();
         // TODO(rkn): Define this constant somewhere else.
         std::string type = "worker_died";
         std::ostringstream error_message;
         error_message << "A worker died or was killed while executing task " << task_id
                       << ".";
         RAY_CHECK_OK(gcs_client_->error_table().PushErrorToDriver(
             job_id, type, error_message.str(), current_time_ms()));
+
+        // Handle the task failure in order to raise an exception in the
+        // application.
+        TreatTaskAsFailed(spec);
+
+        // For dead actors, if there are remaining tasks for this actor, we
+        // should handle them.
+        const ActorID actor_id = worker->GetActorId();
+        if (!actor_id.is_nil()) {
+          CleanUpTasksForDeadActor(actor_id);
```

Comment:
I think we should do this in the `!actor_id.is_nil()` block below instead, since we should do this even if there was no task assigned.

Reply:
good point